lib/sqlalchemy/dialects/mysql/expression.py

```diff
@@ -0,0 +1,138 @@
+from functools import wraps
+
+from sqlalchemy import exc
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.elements import (
+    ColumnElement,
+    ClauseElementBatch,
+)
+
+
+def property_enables_flag(flag_name):
+    def wrapper(target):
+        @property
+        @wraps(target)
+        def inner(self):
+            new_flags = self.flags.copy()
+            new_flags[flag_name] = True
+
+            return match(
+                self.clause,
+                against=self.against,
+                flags=new_flags,
+            )
+
+        return inner
+    return wrapper
+
+
+class match(ColumnElement):
+    """Produce a ``MATCH (X, Y) AGAINST ('TEXT')`` clause.
+
+    E.g.::
```

Comment:
the docs here can be stated in terms of `op.match(x, mysql_opt=...)` and can be part of the mysql/base.py documentation , as an additional bullet at https://docs.sqlalchemy.org/en/14/dialects/mysql.html#mysql-mariadb-sql-extensions or in a new section "MySQL / MariaDB-specific MATCH options", i.e. similar to  https://docs.sqlalchemy.org/en/14/dialects/mysql.html#mysql-mariadb-specific-index-options.

Reply:
done(?)